### DISCUSSION TITLE
Lesson5: failed to install brownie

### DISCUSSION THREAD

**Discussion by Forealthy** [smartcontractkit/full-blockchain-solidity-course-py#1621] [Q&A] [answered]
I have already installed pipx, and finished "python3 -m pip install --user pipx" and "python3 -m pipx ensurepath", but failed to install brownie by "pipx install eth-brownie".  
I examine my pipx:
![image](https://user-images.githubusercontent.com/109839460/182108361-6183e59b-31b3-4c8a-8d11-a02431182325.png)
I think "pipx" has been installed successfully, and then I write "pipx install eth-brownie", but it throws error:
![error_pipx_eth-brownie](https://user-images.githubusercontent.com/109839460/182107597-817b68a5-1c1f-4974-9910-16ce8f6e0ece.png)
Can anyone help? Thanks a lot!

**Comment by cromewar** [comment, marked as answer]
Hello @Forealthy 
This might be the cytools error but not sure as I cannot see the detailed log, anyways... using pipx is not a good choice, specially not on windows (vscode will also find trouble recognizing packages installed with pipx) the best option here is to use just regular `pip` instead of `pipx`.

Also is highly recommendable to set up a python isolated instance with `virtualenv` library.

Cheers.
